ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -214,146 +218,77 @@ impl<'a> SegmentEntry<'a> {
 /// such that this does not count as malicious behavior.
 #[derive(Clone, Debug)]
 struct UnverifiedValuesStore {
-    offset: i64,
-    chunk_size: usize,
-    segment_size: Option<usize>,
-    vec: Vec<Option<UnverifiedValues>>,
+    first_record_of_batch: RecordId,
+    last_record_of_batch: RecordId,
+    multiplication_amount: usize,
+    multiplication_bit_size: usize,
+    is_empty: bool,
+    vec: Vec<UnverifiedValues>,
 }
 
 impl UnverifiedValuesStore {
-    /// Creates a new store for given `chunk_size`. `offset` is initialized to `0`.
-    /// Lazy allocation of the vector. It is allocated once the first segment is added.
-    /// `segment_size` is set to `None` since it may not be known yet
-    fn new(chunk_size: usize) -> Self {
+    /// Creates a new store.
+    /// `first_record_id_of_batch` and `last_record_of_batch` is initialized to `0`.
+    /// The size of the allocated vector is `(amount_of_multiplications * multiplication_bit_size + 255) / 256`
+    fn new(amount_of_multiplications: usize, multiplication_bit_size: usize) -> Self {
         Self {
-            offset: 0,
-            chunk_size,
-            segment_size: None,
-            vec: Vec::<Option<UnverifiedValues>>::new(),
+            first_record_of_batch: RecordId::FIRST,
+            last_record_of_batch: RecordId::FIRST,
+            multiplication_amount: amount_of_multiplications,
+            multiplication_bit_size,
+            is_empty: false,
+            vec: vec![
+                UnverifiedValues::default();
+                (amount_of_multiplications * multiplication_bit_size + 255) >> BIT_ARRAY_SHIFT
+            ],
         }
     }
 
-    /// `update` updates the current store to the next offset which is the previous offset plus `vec.actual_len()`.
-    /// It deallocates `vec`. `chunk_size` remains the same.
-    /// `update` does nothing when `self` is empty, i.e. `vec` is empty.
-    ///
-    /// ## Panics
-    /// Panics when `segment_size` is `None`
+    /// `update` updates the current store to the next chunk of records.
+    /// It sets `last_record_of_batch` and `first_record_of_batch` to the record that follows `last_record_of_batch`.
     fn update(&mut self) {
```

Comment:
I don't think "update" is a very descriptive name for what this method does. This seems more like a "flush" or "reset" function that is only called when a batch is processed. Can you propose a new name?